CRM/Gpapi/Processor.php

```diff
@@ -399,5 +399,26 @@ public static function updateActivityWithUTM(array $params, $activity_id) {
       civicrm_api3('Activity', 'create', $utm_params);
     }
   }
-}
 
+  /**
+   * Resolves a CiviCRM contact ID by
+   *  "identity tracker extension" (if installed).
+   *
+   * @param $contact_id
+   *
+   * @throws \CiviCRM_API3_Exception
+   */
+  public static function identifyContactID(&$contact_id) {
+    if (function_exists('identitytracker_civicrm_install')) {
+      // identitytracker is enabled
+      $contacts = civicrm_api3('Contact', 'findbyidentity', array(
+        'identifier_type' => 'internal',
+        'identifier'      => $contact_id));
+      if ($contacts['count'] == 1) {
+        $contact_id = $contacts['id'];
+        return;
+      }
+    }
+    $contact_id = 0;
```

Comment:
We can be a bit more defensive here and proceed with the original contact id if `identitytracker_civicrm_install` doesn't exist. Technically, we don't have a hard-dependency on `de.systopia.identitytracker`.